build.gradle.kts

```diff
@@ -0,0 +1,95 @@
+import org.gradle.api.tasks.testing.logging.TestLogEvent
+
+plugins {
+  wrapper
+  groovy
+  `java-gradle-plugin`
+  `maven-publish`
+  id("com.fizzpod.sweeney") version ("4.0.0")
+  id("com.github.ksoichiro.build.info") version "0.2.0"
+  id("com.gradle.plugin-publish") version "0.10.1"
+}
+
+sweeney {
+  enforce(mapOf("type" to "jdk", "expect" to "[1.8,)"))
+  enforce("gradle:[5.0,)")
+}
+
+repositories {
+  jcenter()
+}
+
+dependencies {
+  val junitVersion: String by project
+
+  implementation(platform("org.junit:junit-bom:$junitVersion"))
```

Comment:
I've inlined the junit version. Since junit dependencies are managed by that bom, it made the version in gradle.properties not so relevant.

As for the bom import, in this case doesn't really matter whether it sits in the implementation or testImplementation configuration (unless you wanted a junit dependency from being added by mistake to implementation).

Declaring the platform definitions `implementation(platform("org.junit:junit-bom:$junitVersion"))` does not necessarily add those dependencies to the project, it only tells gradle how to resolve the constraints declared by that BOM (similar to what happens with the Maven dependency management).

The actually dependencies are not included in the project until they're added in the dependencies block:

```
dependencies {
  ...
   testImplementation("org.junit.jupiter:junit-jupiter-api")
   testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine")
 ...
}
```

I've added the BOM import to the test implementation as it shouldn't constraint junit dependencies but in the test implementation.

Reply:
👍🏾